assets/src/edit-story/components/canvas/meta/index.js

```diff
@@ -32,45 +55,104 @@ const Options = styled.div`
 const Divider = styled.span`
 	background-color: ${ ( { theme } ) => theme.colors.fg.v1 };
 	opacity: .3;
-	height: 24px;
+	height: ${ HEIGHT }px;
 	width: 1px;
 `;
 
 const Space = styled.div`
-	width: 1em;
+	width: ${ ( { isDouble } ) => isDouble ? 20 : 10 }px;
 `;
 
-const Icon = styled.a`
+const Icon = styled.button`
 	cursor: pointer;
-	color: ${ ( { theme } ) => theme.colors.fg.v4 };
+	background: transparent;
+	border: 0;
+	padding: 0;
+	display: block;
+	color: ${ ( { theme } ) => theme.colors.fg.v1 };
 
-	&:hover {
-		color: ${ ( { theme } ) => theme.colors.fg.v1 };
+	${ ( { disabled } ) => disabled && `
+		opacity: .3;
+		pointer-events: none;
+	` }
+
+	svg {
+		width: 28px;
+		height: 28px;
+		display: block;
 	}
 `;
 
 function Canvas() {
-	const { state: { currentPageNumber, currentPageId }, actions: { deletePage } } = useStory();
+	const { state: { canUndo, canRedo }, actions: { undo, redo } } = useHistory();
+	const { state: { currentPageNumber, currentPage }, actions: { deleteCurrentPage, addPage } } = useStory();
+
+	const handleDeletePage = useCallback(
+		() => deleteCurrentPage(),
+		[ deleteCurrentPage ],
+	);
+
+	const handleAddPage = useCallback(
+		() => addPage( { page: createPage() } ),
+		[ addPage ],
+	);
+
+	const handleDuplicatePage = useCallback(
+		() => addPage( { page: createPage( currentPage ) } ),
+		[ addPage, currentPage ],
+	);
+
+	const handleUndo = useCallback(
+		() => undo(),
+		[ undo ],
+	);
+
+	const handleRedo = useCallback(
+		() => redo(),
+		[ redo ],
+	);
 
 	return (
-		<Box>
-			<PageCount>
-				{ `Page ${ currentPageNumber }` }
-			</PageCount>
-			<Options>
-				<Switch label="Helper" />
-				<Space />
-				<Divider />
-				<Space />
-				<Icon onClick={ () => deletePage( { pageId: currentPageId } ) }>
-					<Delete />
-				</Icon>
-				<Space />
-				<Icon>
-					<Duplicate />
-				</Icon>
-			</Options>
-		</Box>
+		<Wrapper>
+			<Box>
+				<Options>
+					<PageCount>
+						{ `Page ${ currentPageNumber }` }
+					</PageCount>
+					<Space />
+					<Icon onClick={ handleDeletePage }>
+						<Delete />
+					</Icon>
+					<Space />
+					<Icon onClick={ handleDuplicatePage }>
+						<Duplicate />
+					</Icon>
+					<Space />
+					<Icon onClick={ handleAddPage }>
+						<Add />
+					</Icon>
+					<Space />
+					<Divider />
+					<Space />
+					<Icon disabled={ ! canRedo } onClick={ handleRedo }>
+						<Redo />
+					</Icon>
+					<Space />
+					<Icon disabled={ ! canUndo } onClick={ handleUndo }>
+						<Undo />
```

Comment:
I've asked for clarification on Figma, but I assume the position of the undo and redo arrows should probably be switched here.